services/supervisor/tests/supervisor/config.rs

```diff
@@ -0,0 +1,818 @@
+    let signed_confirm = ConfigVote {
+        propose_hash: proposal_hash,
+    }
+    .sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1);
+    testkit.create_block_with_transaction(signed_confirm);
+
+    testkit.create_blocks_until(cfg_change_height.next());
+    assert_eq!(config_propose_entry(&testkit), None);
+    assert_eq!(testkit.consensus_config(), base_consensus_config);
+}
+
+#[test]
+fn test_apply_config_by_min_required_majority() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_validators(4)
+        .with_service(Supervisor)
+        .create();
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+
+    let cfg_change_height = Height(3);
+    let consensus_config = consensus_config_propose_first_variant(&testkit);
+    let config_proposal = ConfigProposeConstructor::new(cfg_change_height)
+        .extend_consensus_config_propose(consensus_config.clone())
+        .config_propose();
+    let proposal_hash = config_proposal.object_hash();
+
+    testkit.create_block_with_transaction(sign_config_propose_transaction(
+        &testkit,
+        config_proposal,
+        initiator_id,
+    ));
+
+    let confirm = ConfigVote {
+        propose_hash: proposal_hash,
+    };
+    // Sign and send confirmation transaction by second validator
+    let keys = testkit.network().validators()[1].service_keypair();
+    testkit.create_block_with_transaction(confirm.clone().sign(
+        SUPERVISOR_INSTANCE_ID,
+        keys.0,
+        &keys.1,
+    ));
+
+    // Sign confirmation transaction by third validator
+    let keys = testkit.network().validators()[2].service_keypair();
+    testkit.create_block_with_transaction(confirm.sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1));
+
+    assert_eq!(config_propose_entry(&testkit), None);
+    assert_eq!(testkit.consensus_config(), consensus_config);
+}
+
+#[test]
+fn test_send_confirmation_by_initiator() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_validators(4)
+        .with_service(Supervisor)
+        .create();
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+
+    let consensus_config = consensus_config_propose_first_variant(&testkit);
+    let config_proposal = ConfigProposeConstructor::new(CFG_CHANGE_HEIGHT)
+        .extend_consensus_config_propose(consensus_config.clone())
+        .config_propose();
+    let proposal_hash = config_proposal.object_hash();
+
+    testkit.create_block_with_transaction(sign_config_propose_transaction(
+        &testkit,
+        config_proposal,
+        initiator_id,
+    ));
+
+    // Try to send confirmation transaction by the initiator
+    let keys = testkit.network().us().service_keypair();
+    let signed_confirm = ConfigVote {
+        propose_hash: proposal_hash,
+    }
+    .sign(SUPERVISOR_INSTANCE_ID, keys.0, &keys.1);
+    testkit
+        .create_block_with_transaction(signed_confirm)
+        .transactions[0]
+        .status()
+        .unwrap_err();
```

Comment:
Nit: I think `expect_err` or even a check for expected error will be more informative.